cirq/contrib/noise_models/noise_models.py

```diff
@@ -173,16 +174,136 @@ def noisy_moment(self, moment: 'cirq.Moment',
+            return moments
+
+
+def simple_noise_from_calibration_metrics(calibration: engine.Calibration,
+                                          depolNoise: bool = False,
+                                          dampingNoise: bool = False,
+                                          readoutDecayNoise: bool = False,
+                                          readoutErrorNoise: bool = False
                                          ) -> devices.NoiseModel:
-    """Creates a reasonable PerQubitDepolarizingNoiseModel using the provided
-    calibration data. This object can be retrived from the engine by calling
-    'get_latest_calibration()' or 'get_calibration()' using the ID of the
-    target processor.
+    """Creates a reasonable PerQubitDepolarizingWithDampedReadoutNoiseModel
+    using the provided calibration data.
+
+    Args:
+        calibration: a Calibration object (cirq/google/engine/calibration.py).
+            This object can be retrived from the engine by calling
+            'get_latest_calibration()' or 'get_calibration()' using the ID of
+            the target processor.
+        depolNoise: Enables per-gate depolarization if True.
+        dampingNoise: Enables per-gate amplitude damping if True.
+            Currently unimplemented.
+        readoutDecayNoise: Enables pre-readout amplitude damping if True.
+        readoutErrorNoise: Enables pre-readout bitflip errors if True.
+
+    Returns:
+        A PerQubitDepolarizingWithDampedReadoutNoiseModel with error
+            probabilities generated from the provided calibration data.
     """
+    if not any([depolNoise, dampingNoise, readoutDecayNoise, readoutErrorNoise
+               ]):
+        raise ValueError('At least one error type must be specified.')
     assert calibration is not None
-    rb_data: Dict['cirq.Qid', float] = {
-        qubit[0]: depol_prob[0] for qubit, depol_prob in
-        calibration['single_qubit_rb_total_error'].items()
-    }
-    return PerQubitDepolarizingNoiseModel(rb_data)
+    depol_prob_map: Dict['cirq.Qid', float] = {}
+    readout_decay_map: Dict['cirq.Qid', float] = {}
+    readout_error_map: Dict['cirq.Qid', float] = {}
+
+    if depolNoise:
+        depol_prob_map = {
+            qubit[0]: depol_prob[0] for qubit, depol_prob in
+            calibration['single_qubit_rb_total_error'].items()
+        }
+    if dampingNoise:
+        # TODO: implement per-gate amplitude damping noise.
+        raise NotImplementedError('Gate damping is not yet supported.')
+
+    if readoutDecayNoise:
+        # Copied from Sycamore readout duration in known_devices.py
+        # TODO: replace with polling from DeviceSpecification.
+        readout_micros = 1
+        readout_decay_map = {
+            qubit[0]: exp(1 - readout_micros / t1[0])
+            for qubit, t1 in calibration['single_qubit_idle_t1_micros'].items()
+        }
+    if readoutErrorNoise:
+        # This assumes that p(<1|0>) is negligible for readout timescales.
```

Comment:
Is this correct?  Isn't the probability that a state decays from |1> to |0> generally higher than measuring |1> if it was supposed to be |0>?

Shouldn't this use p1 error somehow?  Is the intention to use p1 in the amplitude damping condition?

Reply:
The intent is to model decay from `|1>` to `|0>` - if I've got my notation correct, `p(<1|0>)` is the probability of going from `|0>` to `|1>`.

Is `single_qubit_readout_p0_error` the right metric for `|1>` to `|0>` decay? If we have a metric for the `|0>` to `|1>` error, I could add that as well - it would require replacing the AmplitudeDamping channel with a GAD channel in the model.

Reply:
The p0 (or p00) experiment is to prepare the state in |0> and try to measure |0>.  The error is the chance that this doesn't happen.
The p1 (or p11) experiment is to prepare the state in |1> and try to measure |1>.  The error is the chance that this doesn't happen.

So, I think you want to p1 error if you are trying to capture the probability that the state will decay from |1> to |0> either by incorrect measurement, incorrect state preparation, decay, or what-have-you.

Reply:
Whoops, I think I misread my own code. `readout_error_map` should capture error in the readout, not from qubit decay. Reading the doc in #2761, it seems like `single_qubit_readout_separation_error` is actually what we want here.

(T1 decay during readout is captured by the aptly-named `readout_decay_map`.)

Fix in progress.

Reply:
Done. Also switched to `single_qubit_rb_pauli_error` for depolarization, since that appears to be more precise than using total error.